packages/webapp/src/containers/Finances/selectors.js

```diff
@@ -23,7 +24,33 @@ const selectedSaleSelector = createSelector(financeSelector, (state) => state.se
 
 const expenseSelector = createSelector(financeSelector, (state) => state.expenses);
 
-const expenseTypeSelector = createSelector(financeSelector, (state) => state.expense_types);
+const allExpenseTypeSelector = createSelector(financeSelector, (state) => state.expense_types);
+
+const expenseTypeSelector = createSelector(financeSelector, (state) => {
+  return state.expense_types?.filter((type) => !type.deleted);
+});
+
+const expenseTypeTileContentsSelector = createSelector(financeSelector, (state) => {
```

Comment:
I think I see why this logic has to go here (since the class-based React container means you can't useMemo and put it in the container as done in taskTypes), but is there another reason to prefer using a selector here? Specifically for custom revenue could this filtering logic go back into the container in a useMemo hook? Or should it go into the slice?

Reply:
I have been wondering if selector should do this these days! (it will not be just a selector...)
I just followed the pattern of `filteredTaskCardContentSelector` here without taking class/functional component into account.
https://github.com/LiteFarmOrg/LiteFarm/blob/78cd987dc9846c439c31f9c14b2b6092640acdd2/packages/webapp/src/containers/Task/taskCardContentSelector.js#L72-L79

We could create a hook like `useCropCatalogue` (extract useMemo that you mentioned)
https://github.com/LiteFarmOrg/LiteFarm/blob/integration/packages/webapp/src/containers/CropCatalogue/useCropCatalogue.js

For custom revenue, I think the logic can be written in the container.

Reply:
Okay thank you 🙏  Yes our crashing selectors have gotten me thinking a lot recently about if less logic can go into them...!

Reply:
Same!!!!!